src/app/modules/item/components/thread/thread.component.ts

```diff
@@ -1,25 +1,68 @@
-import { Component } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, OnDestroy, ViewChild } from '@angular/core';
 import { ThreadService } from '../../services/threads.service';
+import { FormBuilder } from '@angular/forms';
+import { readyData } from '../../../../shared/operators/state';
+import { Subscription, combineLatest } from 'rxjs';
+import { UserSessionService } from '../../../../shared/services/user-session.service';
 
 @Component({
   selector: 'alg-thread',
   templateUrl: './thread.component.html',
   styleUrls: [ './thread.component.scss' ],
 })
-export class ThreadComponent {
-  messageToSend = '';
+export class ThreadComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('messagesScroll') messagesScroll?: ElementRef<HTMLDivElement>;
+
+  form = this.fb.nonNullable.group({
+    messageToSend: [ '' ],
+  });
 
   state$ = this.threadService.state$;
 
+  private subscription?: Subscription;
+
   constructor(
     private threadService: ThreadService,
+    private fb: FormBuilder,
+    private userSessionService: UserSessionService,
   ) {}
 
+  ngAfterViewInit(): void {
+    this.subscription = combineLatest([
+      this.state$.pipe(readyData()),
+      this.userSessionService.userProfile$,
+    ]).subscribe(([ events, currentUser ]) => {
+      const lastEvent = events[events.length - 1];
+      if (lastEvent && lastEvent.createdBy === currentUser.groupId) {
+        setTimeout(() => {
+          this.scrollDown();
+        });
+      }
+    });
```

Comment:
It means, you only scroll down if the new message is yours.
But we may consider also that anyone on the bottom of the scroll (the default behavior) just want to keep staying at the bottom and see new messages. So I think the rule should be instead: "if the user is already at the bottom of the scroll, keep him at the bottom of the scroll by scrolling when there are new events".

Reply:
Fixed